keel-artifact/src/main/kotlin/com/netflix/spinnaker/keel/artifacts/ArtifactListener.kt

```diff
@@ -148,6 +146,21 @@ class ArtifactListener(
   private fun getLatestStoredVersion(artifact: DeliveryArtifact): String? =
     repository.artifactVersions(artifact).sortedWith(artifact.versioningStrategy.comparator).firstOrNull()
 
+  /**
+   * Returns a copy of the [PublishedArtifact] with the git and build metadata populated, if available.
+   */
+  private fun ArtifactSupplier<*,*>.addMetadata(artifact: PublishedArtifact): PublishedArtifact {
+    val artifactMetadata = runBlocking {
+      try {
+        getArtifactMetadata(artifact)
+      } catch (ex: Exception) {
+        log.error("Could not fetch artifact metadata for name ${artifact.name} and version ${artifact.version}", ex)
+        null
+      }
+    }
+    return artifact.copy(gitMetadata = artifactMetadata?.gitMetadata, buildMetadata = artifactMetadata?.buildMetadata)
```

Comment:
we should only add the metadata if it's not null. otherwise, let's just return the original artifact

Reply:
I didn't think it was worth the extra `if` statement because the original object will also have those fields set to `null` by default...

Reply:
Are you sure those fields will be set to null?